app/src/main/java/protect/card_locker/LoyaltyCardListDisplayOptionsManager.java

```diff
@@ -0,0 +1,177 @@
+package protect.card_locker;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class LoyaltyCardListDisplayOptionsManager {
+    public final Context mContext;
+
+    private final Runnable mRefreshCardsCallback;
+    private final Runnable mSwapCursorCallback;
+
+    protected SharedPreferences mCardDetailsPref;
+
+    private boolean mShowNameBelowThumbnail;
+    private boolean mShowNote;
+    private boolean mShowBalance;
+    private boolean mShowValidity;
+    private boolean mShowArchivedCards;
+
+    public static class LoyaltyCardDisplayOption {
+        public String name;
+        public boolean value;
+        public Consumer<Boolean> callback;
+
+        LoyaltyCardDisplayOption(String name, boolean value, Consumer<Boolean> callback) {
+            this.name = name;
+            this.value = value;
+            this.callback = callback;
+        }
+    }
+
+    public LoyaltyCardListDisplayOptionsManager(Context context, @NonNull Runnable refreshCardsCallback, @Nullable Runnable swapCursorCallback) {
+        mContext = context;
+        mRefreshCardsCallback = refreshCardsCallback;
+        mSwapCursorCallback = swapCursorCallback;
+
+        // Retrieve user details preference
+        mCardDetailsPref = mContext.getSharedPreferences(
+                mContext.getString(R.string.sharedpreference_card_details),
+                Context.MODE_PRIVATE);
+        mShowNameBelowThumbnail = mCardDetailsPref.getBoolean(mContext.getString(R.string.sharedpreference_card_details_show_name_below_thumbnail), false);
+        mShowNote = mCardDetailsPref.getBoolean(mContext.getString(R.string.sharedpreference_card_details_show_note), true);
+        mShowBalance = mCardDetailsPref.getBoolean(mContext.getString(R.string.sharedpreference_card_details_show_balance), true);
+        mShowValidity = mCardDetailsPref.getBoolean(mContext.getString(R.string.sharedpreference_card_details_show_validity), true);
+        mShowArchivedCards = mCardDetailsPref.getBoolean(mContext.getString(R.string.sharedpreference_card_details_show_archived_cards), true);
```

Comment:
Are archived cards supposed to be shown by default?

Reply:
Yeah, they're sorted at the bottom of the list. I want to avoid the situation where people archive their cards and then can't find them anymore, thus showing them by default.

Reply:
Makes sense. Just wanted to make sure.